dotCMS/src/main/java/com/dotmarketing/util/ImportUtil.java

```diff
@@ -814,46 +766,42 @@ else if(UtilMethods.isSet(y)){
             }
 
             //Find the relationships and their related contents
-            HashMap<Relationship,List<Contentlet>> csvRelationshipRecordsParentOnly = new HashMap<Relationship,List<Contentlet>>();
-            HashMap<Relationship,List<Contentlet>> csvRelationshipRecordsChildOnly = new HashMap<Relationship,List<Contentlet>>();
-            HashMap<Relationship,List<Contentlet>> csvRelationshipRecords = new HashMap<Relationship,List<Contentlet>>();
+            HashMap<Relationship, List<Contentlet>> csvRelationshipRecordsParentOnly = new HashMap<>();
+            HashMap<Relationship, List<Contentlet>> csvRelationshipRecordsChildOnly = new HashMap<>();
+            HashMap<Relationship, List<Contentlet>> csvRelationshipRecords = new HashMap<>();
             for (Integer column : relationships.keySet()) {
                 Relationship relationship = relationships.get(column);
                 String relatedQuery = line[column];
-                List<Contentlet> relatedContentlets = new ArrayList<Contentlet>();
+                List<Contentlet> relatedContentlets = new ArrayList<>();
                 boolean error = false;
-                if(UtilMethods.isSet(relatedQuery))
-                {
+                if (UtilMethods.isSet(relatedQuery)) {
                     relatedContentlets = conAPI.checkoutWithQuery(relatedQuery, user, false);
 
                     //validate if the contenlet retrieved are from the correct typ
-                    if(FactoryLocator.getRelationshipFactory().isParent(relationship,contentType))
-                    {
-                        for(Contentlet contentlet : relatedContentlets)
-                        {
+                    if (FactoryLocator.getRelationshipFactory()
```

Comment:
double check if this method is closing the db here

Reply:
Looks safe